workers/unity/Packages/com.improbable.gdk.tools/DevAuthTokenUtils.cs

```diff
@@ -33,18 +43,53 @@ private static void Generate()
 
             try
             {
-                if (Json.Deserialize(receivedMessage).TryGetValue(JsonDataKey, out var jsonData) &&
-                    ((Dictionary<string, object>) jsonData).TryGetValue(TokenSecretKey, out var tokenSecret))
+                var deserializedMessage = Json.Deserialize(receivedMessage);
+                if (deserializedMessage.TryGetValue(JsonDataKey, out var jsonData) &&
+                    ((Dictionary<string, object>) jsonData).TryGetValue(JsonTokenSecretKey, out var tokenSecret))
                 {
                     devAuthToken = (string) tokenSecret;
                 }
+                else
+                {
+                    throw new Exception(
+                        $@"{(deserializedMessage.TryGetValue(JsonErrorKey, out var errorMessage)
```

Comment:
This looks like a bit overkill to all put in a string format, maybe pull the logic out?